headers.go

```diff
@@ -62,6 +62,19 @@ func (h ProtectedHeader) SetAlgorithm(alg Algorithm) {
 	h[HeaderLabelAlgorithm] = int(alg)
 }
 
+// GetAlgorithm gets the algorithm value from the algorithm header.
+func (h ProtectedHeader) GetAlgorithm() (Algorithm, error) {
+	value, ok := h[HeaderLabelAlgorithm]
+	if !ok {
+		return 0, ErrAlgorithmNotFound

```

Comment:
[RFC 8152 Section 16.4](https://www.rfc-editor.org/rfc/rfc8152.html#section-16.4) says the value 0 is reserved, can't be used. With this in mind, does it makes sense to not return an error here, just `return 0, nil`? I'm saying this because it is not mandatory for protected headers to contain an algorithm parameter, so it seem weird to return an error.

Reply:
The value `0` is valid value even if it is reserved. Therefore, we should return an error for that.

```go
alg, err := header.GetAlgorithm()
if err == nil {
   // do stuff with the alg
} else if err != ErrAlgorithmNotFound {
   panic(err)
} else {
  // alg does not present.
}
```

With above code, the caller can know whether the `alg` header present or not not.

If we `return 0, nil`, then "`alg` not present" and "`alg` is `0 `" are not distinguishable.

Reply:
> The value 0 is valid value even if it is reserved. Therefore, we should return an error for that.

That's a pitty. Then the error sentinel is the way to go! 👍